flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/ScalarFunction.scala

```diff
@@ -56,8 +58,10 @@ abstract class ScalarFunction extends UserDefinedFunction {
     ScalarFunctionCall(this, params)
   }
 
-  override def toString: String = getClass.getCanonicalName
-
+  override def toString: String = {
+    val md5  =  DigestUtils.md5Hex(serialize(this))
```

Comment:
I think this is dangerous. 
The function might be overridden by a UDF.
Rather add a new `final` method.